xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp

```diff
@@ -1721,9 +1722,17 @@ CDemuxStream* CDVDDemuxFFmpeg::AddStream(int streamIdx)
               *reinterpret_cast<AVContentLightMetadata*>(side_data));
         }
 
-        AVDictionaryEntry* rtag = av_dict_get(pStream->metadata, "rotate", NULL, 0);
-        if (rtag)
-          st->iOrientation = atoi(rtag->value);
+        uint8_t* displayMatrixSideData =
+            av_stream_get_side_data(pStream, AV_PKT_DATA_DISPLAYMATRIX, nullptr);
+        if (displayMatrixSideData)
+        {
+          const double tetha =
+              av_display_rotation_get(reinterpret_cast<int32_t*>(displayMatrixSideData));
+          if (!std::isnan(tetha))
+          {
+            st->iOrientation = ((static_cast<int>(-tetha) % 360) + 360) % 360;
```

Comment:
The angle (in degrees) by which the transformation rotates the frame counterclockwise. The angle will be in range [-180.0, 180.0], or NaN if the matrix is singular.

Reply:
Why do you need the %360?

Reply:
I think (int(-theta) + 360) %360 should be enough, right?

Reply:
Got it from mpv but I agree `(int(-theta) + 360) %360 ` should suffix. Probably they divide first to ensure rounding?

Reply:
```
    sd = av_frame_get_side_data(src, AV_FRAME_DATA_DISPLAYMATRIX);
    if (sd) {
        double r = av_display_rotation_get((int32_t *)(sd->data));
        if (!isnan(r))
            dst->params.rotate = (((int)(-r) % 360) + 360) % 360;
    }
 ```
 
 Yes, indeed. Following the second comment here: https://stackoverflow.com/questions/11720656/modulo-operation-with-negative-numbers I rather think this is the issue. Please keep what mpv had ... sorry for the noise.

Reply:
I see. Thanks for looking into it - i've reverted to the previous version.
A green for merging is welcome :)